packages/arb-token-bridge-ui/src/components/TransactionHistory/helpers.ts

```diff
@@ -133,6 +149,99 @@ function getWithdrawalStatusFromReceipt(
   }
 }
 
+export function getDepositsWithoutStatusesFromCache(): Deposit[] {
+  return JSON.parse(
+    localStorage.getItem(DEPOSITS_LOCAL_STORAGE_KEY) ?? '[]'
+  ) as Deposit[]
+}
+
+/**
+ * Cache deposits from event logs. We don't fetch these so we need to store them locally.
+ *
+ * @param {MergedTransaction} tx - Deposit from event logs to be cached.
+ */
+export function addDepositToCache(tx: Deposit) {
+  if (tx.direction !== 'deposit') {
+    return
+  }
+
+  const cachedDeposits = getDepositsWithoutStatusesFromCache()
+
+  const foundInCache = cachedDeposits.find(cachedTx =>
+    isSameTransaction(
+      { ...cachedTx, txId: cachedTx.txID },
+      { ...tx, txId: tx.txID }
+    )
+  )
+
+  if (foundInCache) {
+    return
+  }
+
+  const newCachedDeposits = [tx, ...cachedDeposits]
+
+  localStorage.setItem(
+    DEPOSITS_LOCAL_STORAGE_KEY,
+    JSON.stringify(newCachedDeposits)
+  )
+}
+
+/**
+ * Cache parent chain tx details when claiming. This is the chain the funds were claimed on. We store locally because we don't have access to this tx from the child chain tx data.
+ *
+ * @param {MergedTransaction} tx - Transaction that initiated the withdrawal (child chain transaction).
+ * @param {string} parentChainTxId - Transaction ID of the claim transaction (parent chain transaction ID).
+ */
+export function setWithdrawalClaimParentChainTxDetails(
```

Comment:
NIT: `setParentChainTxDetailsOfWithdrawalClaimTx` and similarly in other function?

Reply:
updated